test/unit/user/permissions_test.rb

```diff
@@ -150,4 +150,62 @@ class User::PermissionsTest < ActiveSupport::TestCase
     assert user.forbidden_some_services?
   end
 
+  class PermissionForEventTest < ActiveSupport::TestCase
+    setup do
+      @user = User.new
+      @cinstance = FactoryBot.create(:cinstance)
+      @service = cinstance.service
+    end
+
+    attr_reader :user, :cinstance, :service
+
+    test 'Services::ServiceDeletedEvent' do
+      service_deleted_event = Services::ServiceDeletedEvent.create(service)
+      user.expects(:has_access_to_service?).with(service.id).returns(true)
+      assert user.permission_for_event?(service_deleted_event)
+      user.expects(:has_access_to_service?).with(service.id).returns(false)
+      refute user.permission_for_event?(service_deleted_event)
+    end
+
+    test 'Cinstance/Application events checks permission for plans and partners' do
+      [
+        Cinstances::CinstanceCancellationEvent.create(cinstance), Cinstances::CinstanceExpiredTrialEvent.create(cinstance),
+        Cinstances::CinstancePlanChangedEvent.create(cinstance, user), Applications::ApplicationCreatedEvent.create(cinstance, user)
+      ].each do |cinstance_event|
```

Comment:
Can we name the array with a separate variable? Maybe put it in `#setup` (although there's a test that does not need it):

```ruby
@events = [
  Cinstances::CinstanceCancellationEvent.create(cinstance),
  Cinstances::CinstanceExpiredTrialEvent.create(cinstance),
  Cinstances::CinstancePlanChangedEvent.create(cinstance, user),
  Applications::ApplicationCreatedEvent.create(cinstance, user)
]
```